brian2/synapses/synapses.py

```diff
@@ -993,12 +994,12 @@ def _create_variables(self, equations, user_dtype=None):
                 self.variables.add_reference(name + '_pre', self.source, name,
                                              index=index)
             except TypeError:
-                logger.debug(('Cannot include a reference to {var} in '
-                              '{synapses}, {var} uses a non-standard indexing '
-                              'in the pre-synaptic group '
-                              '{source}.').format(var=name,
-                                                  synapses=self.name,
-                                                  source=self.source.name))
+                logger.diagnostic(('Cannot include a reference to {var} in '
+                                   '{synapses}, {var} uses a non-standard indexing '
+                                   'in the pre-synaptic group '
+                                   '{source}.').format(var=name,
+                                                       synapses=self.name,
+                                                       source=self.source.name))
```

Comment:
I am not 100% sure, but I think it covers situations like this:
```Python
group1 = NeuronGroup(5, 'x : 1')
group2 = NeuronGroup(5, 'x_l : 1 (linked)')
group2.x_l = linked_var(group1, 'x', [4, 1, 2, 3, 0])
S = Synapses(group2, group2, ...)
```
In this situation, you cannot use `x_l_pre` or `x_l_post` in the synaptic group, as it would require double indexing. I think in the long run, the best solution would be to include special `UnusableVariable` objects in this case which would allow to give a meaningful error message when the user actually tries to use one of these variables (currently they'll simply get a general "unknown identifier" error message).